dts/arm/st/f7/stm32f723.dtsi

```diff
@@ -15,6 +15,11 @@
 		reg = <0x20010000 DT_SIZE_K(192)>;
 	};
 
+	dtcm: memory@20000000 {
+		compatible = "st,stm32-dtcm";
```

Comment:
The DTCM memory is not specific to the STM32F7 nor to ST, but is found in all Cortex-M7 SoCs. I therefore think we should probably call it something like 'arm,dtcm'.

Reply:
Where is this `Cortex-M7` exactly?